component/xload/comp/data_manager.go

```diff
@@ -0,0 +1,120 @@
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE
+*/
+
+package comp
+
+import (
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
+	"github.com/Azure/azure-storage-fuse/v2/component/xload/common"
+	"github.com/Azure/azure-storage-fuse/v2/internal"
+	"github.com/Azure/azure-storage-fuse/v2/internal/handlemap"
+)
+
+// verify that the below types implement the xcomponent interfaces
+var _ common.XComponent = &dataManager{}
+var _ common.XComponent = &remoteDataManager{}
+
+const DATA_MANAGER string = "DATA_MANAGER"
+
+type dataManager struct {
+	common.XBase
+}
+
+// --------------------------------------------------------------------------------------------------------
+
+type remoteDataManager struct {
+	dataManager
+}
+
+func NewRemoteDataManager(remote internal.Component) (*remoteDataManager, error) {
+	log.Debug("data_manager::NewRemoteDataManager : create new remote data manager")
+
+	rdm := &remoteDataManager{}
+
+	rdm.SetName(DATA_MANAGER)
+	rdm.SetRemote(remote)
+	rdm.Init()
+	return rdm, nil
+}
+
+func (rdm *remoteDataManager) Init() {
+	rdm.SetThreadPool(common.NewThreadPool(common.MAX_WORKER_COUNT, rdm.Process))
+	if rdm.GetThreadPool() == nil {
+		log.Err("remoteDataManager::Init : fail to init thread pool")
+	}
+}
+
+func (rdm *remoteDataManager) Start() {
+	log.Debug("remoteDataManager::Start : start remote data manager")
+	rdm.GetThreadPool().Start()
+}
+
+func (rdm *remoteDataManager) Stop() {
+	log.Debug("remoteDataManager::Stop : stop remote data manager")
+	if rdm.GetThreadPool() != nil {
+		rdm.GetThreadPool().Stop()
+	}
+}
+
+// upload or download block
+func (rdm *remoteDataManager) Process(item *common.WorkItem) (int, error) {
+	if item.Download {
+		return rdm.ReadData(item)
+	} else {
+		return rdm.WriteData(item)
+	}
+}
+
+// ReadData reads data from the data manager
+func (rdm *remoteDataManager) ReadData(item *common.WorkItem) (int, error) {
+	log.Debug("remoteDataManager::ReadData : Scheduling download for %s offset %v", item.Path, item.Block.Offset)
+
+	h := handlemap.NewHandle(item.Path)
```

Comment:
We shall not create dummy handle here, instead pass the path, file-size as input parameter in option bag

Reply:
Done, added `Path` and `Size` parameters in `ReadInBufferOptions` so that we don't need to create a dummy handle.